proto/stride/airdrop/query.proto

```diff
@@ -0,0 +1,103 @@
+
+syntax = "proto3";
+package stride.airdrop;
+
+import "stride/airdrop/airdrop.proto";
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+import "cosmos_proto/cosmos.proto";
+
+option go_package = "github.com/Stride-Labs/stride/v22/x/airdrop/types";
+
+// Query defines the gRPC querier service.
+service Query {
+  // Queries all allocations across all addresses
+  rpc AllAllocations(QueryAllAllocationsRequest)
+      returns (QueryAllAllocationsResponse) {
+    option (google.api.http).get =
+        "/Stride-Labs/stride/airdrop/all_allocations/{id}";
+  }
+
+  // Queries the allocation for a given user for an airdrop
+  rpc UserAllocation(QueryUserAllocationRequest)
+      returns (QueryUserAllocationResponse) {
+    option (google.api.http).get =
+        "/Stride-Labs/stride/airdrop/user_allocation/{id}/{address}";
+  }
+
+  // Queries the allocations for a given user across all airdrops
+  rpc UserAllocations(QueryUserAllocationsRequest)
+      returns (QueryUserAllocationsResponse) {
+    option (google.api.http).get =
+        "/Stride-Labs/stride/airdrop/user_allocations/{address}";
+  }
+
+  // Queries the state of an address for an airdrop (daily claim, claim & stake,
+  // upfront)
+  rpc UserClaimType(QueryUserClaimTypeRequest)
+      returns (QueryUserClaimTypeResponse) {
+    option (google.api.http).get =
+        "/Stride-Labs/stride/airdrop/user_claim_type/{id}/{address}";
+  }
```

Comment:
do you think we need a separate query for this? I was thinking we could maybe rework this into a `UserSummary` query that returns:
* Claim type 
* Amount claimed so far
* Amount remaining

Then that would be the main frontend query and we could use the `UserAllocations` more for auditing/introspection

Reply:
This is just `UserState` from the spec doc. Feel free to change this.